itou/approvals/models.py

```diff
@@ -564,6 +564,21 @@ class Meta:
                 """,
                 declare=[("current_employee_record_id", "INT")],
             ),
+            pgtrigger.Trigger(
+                name="plan_pe_notification_on_date_updates",
+                when=pgtrigger.Before,
+                operation=pgtrigger.UpdateOf("start_at", "end_at"),
+                condition=pgtrigger.Q(old__end_at__df=pgtrigger.F("new__end_at"))
+                | pgtrigger.Q(old__start_at__df=pgtrigger.F("new__start_at")),
+                func=f"""
+                    -- If there is an "UPDATE" action on 'approvals_approval' table (Approval model object)
+                    -- and dates are modified then reset pe_notification_status field to PENDING status
+                    IF (TG_OP = 'UPDATE') THEN
+                        NEW.pe_notification_status := '{api_enums.PEApiNotificationStatus.PENDING}';
+                    END IF;
```

Comment:
Il me semble que le `IF` n'est pas nécessaire puisque c'est seulement sur les `UPDATE` à cause du `UpdateOf`.

Par contre ça me rappelle la discussion au séminaire à propos de l'historique des champs, ne faudrait-il pas ajouter un  `RAISE EXCEPTION` avant le `:=` dans le cas où `NEW.pe_notification_status IS DISTINCT FROM OLD.pe_notification_status` pour ne pas _shadow_ un changement depuis le code ?